### PR TITLE
Fix error reading concatenated pdfs

### PR DESCRIPTION
**Pull request by jordan-thoms**
This fixes issue #5909

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
It's *great* that you've included a test-case in the patch, but looking at it I'm not sure that we're actually allowed to include (and redistribute) the PDF file in the PDF.js repo. At the top of the pages, it says:
> This document is being provided for the exclusive use of HONG YUN YONG at [FIRM]. Not for redistribution."

Would you be able to create a reduced test-case, without any potentially private data?

**Comment by jordan-thoms** [comment]
I tried to and didn't manage to create quite the same problem - I created one which exhibited the same behavior in PDF.js, but it caused Adobe Reader to segfault (whereas Adobe reads the one I attached fine). I think it has to do with the object IDs in the two files, and if they are overlapping.

I'll try again to create one which has the same behavior, perhaps taking the current PDF and removing some of the streams.

**Comment by jordan-thoms** [comment]
I've replaced it with a new test file. It's got the same structure, but I've snipped out the content streams.

File: test/test_manifest.json [hunk 1]
```diff
@@ -2080,5 +2080,11 @@
        "md5": "d5b37f8bf1b3aafa1b4fcf19ebdc7c74",
        "rounds": 1,
        "type": "eq"
+    },
+    {  "id": "issue5909",
+       "file": "pdfs/issue5909.pdf",
+       "md5": "65c169b6f540b27ac0ff2738a80d1e14",
+       "rounds": 1,
+       "type": "eq"
```
Comment (Snuffleupagus) [inline test/test_manifest.json]: I think you can change this to `load` test instead.

Comment (jordan-thoms) [inline test/test_manifest.json]: Ok, yeah I can seeing as it's just a blank PDF - it's not going to render anything anyway. I'll make that change and push up a new commit.

**Comment by jordan-thoms** [comment]
Updated the test to just a load test.

**Comment by jordan-thoms** [comment]
Looks like the windows failure is false, some file locking issue?

**Comment by timvandermeij** [comment]
@jordan-thoms Yes, that is not your fault. It has to be corrected on the Windows bot.

File: test/test_manifest.json [hunk 2]
```diff
@@ -2080,5 +2080,11 @@
        "md5": "d5b37f8bf1b3aafa1b4fcf19ebdc7c74",
        "rounds": 1,
        "type": "eq"
+    },
+    {  "id": "issue5909",
+       "file": "pdfs/issue5909.pdf",
+       "md5": "65c169b6f540b27ac0ff2738a80d1e14",
```
Comment (Snuffleupagus) [inline test/test_manifest.json]: The bots complain:
```
WARNING: MD5 of file "pdfs/issue5909.pdf" does not match file. Expected "65c169b6f540b27ac0ff2738a80d1e14" computed "51a724136c0c10008bd061a78ea4b8fc"
```

Comment (timvandermeij) [inline test/test_manifest.json]: @jordan-thoms Perhaps you forgot to update the MD5 hash after updating the test file?

Comment (jordan-thoms) [inline test/test_manifest.json]: oh yes. I'll push up a new commit.

**Comment by timvandermeij** [comment]
Nice, thank you!

**Pull request merged**